packages/@lwc/synthetic-shadow/src/polyfills/event/polyfill.ts

```diff
@@ -74,14 +83,21 @@ function patchedTargetGetter(this: Event): EventTarget | null {
 
 function patchedComposedPathValue(this: Event): EventTarget[] {
     const originalTarget = eventTargetGetter.call(this);
-    const originalCurrentTarget = eventCurrentTargetGetter.call(this);
 
     // Account for events with targets that are not instances of Node (e.g., when a readystatechange
     // handler is listening on an instance of XMLHttpRequest).
     if (!(originalTarget instanceof Node)) {
         return [];
     }
 
```

Comment:
```suggestion
    // If the original target is inside a native shadow root, then just call the native `composePath()` method.
    // The path is already retargeted, and the native shadow root can't contain any synthetic shadow roots
    // (because we don't allow synthetic-within-native), so there's nothing to do.
```